### ISSUE TITLE
Question: Is nesting in same type overlapping/stacked DND zones possible?

### ISSUE THREAD

**Issue by PaulBeaudet** [isaacHagoel/svelte-dnd-action#254]
This idea might be breaking an assumption of the library. 

Say we were talking in Trello board terms a series of lists and the containing cards would be two separate DND zone types. This much I understand and could replicate with this library. Where I want to divert from the functionality is the ability to drag a card within a card whilst maintaining the ability to prioritize cards within the list. 

Right now I have the default DND type set up with DND zones within "list" and "card" like components. As soon I pick up a card "on: consider" fires as considering to drop the card within itself. If I drop the card in another an uncaught promise error gets thrown for a missing id property, the past card empties, and the on: finalize populates the card data within the card as undefined. Trying to reprioritize cards throws "Uncaught TypeError: Cannot read property 'isDndShadowItem' of undefined". Both component lists are being rendered with their ids. 

Seems to me that with this type of intended behavior it would be difficult to determine what the right thing to hovered over and part is. I'm not too surprised it doesn't work. All that said my question is, is this type of behavior possible? 

The closest similar example I can see is the "crazy nesting" one. In that example, the author seems to have managed to initialize the DND zone under one component umbrella (that recursively calls itself?). Even in that example, it's not possible to put an item within an item or node within an item, only a node within a node. Not sure I completely understand what's going in that example. Specifically why the "shadow placeholder" needs to be filtered out.

**Comment by isaacHagoel** [comment]
Hi,
if you can create a simple REPL that illustrates what you are trying to do
(it will be broken, i understand), i might be able to fix it for you and
explain what i did.
thx

**Comment by PaulBeaudet** [comment]
Hi Isaac,

Seems like I should take a look at my application a bit more carefully. Doing what I was thinking about in the REPL from scratch was easy enough. I think the dnd zones in my app are right on top of each other. Which might be causing some of the issues explained above.

Here's what I came up with that clearly shows stacked zones are not an issue- https://svelte.dev/repl/de5d48ca7cad41f09a1bb7246212dab6?version=3.35.0

**Comment by PaulBeaudet** [comment]
Actually, on a closer look, that example is throwing a lot of errors. Probably because I'm not updating the parent component data

**Comment by isaacHagoel** [comment]
changing from ul to section in your List component fixed the errors.
you had a ul with other uls as children (cards) which is not legal markup.
https://svelte.dev/repl/4281aedcc02441728e199f4bc264e7ed?version=3.35.0

**Comment by PaulBeaudet** [comment]
Thanks for taking the time to look at this Isaac. I appreciate the work that you've done on this library. It's really helped me quickly get up and running with drag and drop without getting into the details of the dom API. The keyboard/accessibility features are a great touch.

Each card was wrapped in an <li> It was actually <ul>s inside of <li>s. Regardless, it's probably the same markup issue, I do see your point. The section fix takes care of the errors that were being thrown. Still, now that we have the concept working, it's almost impossible to use because the animation moves the nested drop zone away from the cursor while hovering over it. I'm going to try to tweak this REPL example before trying to fix my application in a similar way and report back what works.

**Comment by PaulBeaudet** [comment]
After some more experimenting, seems like there are at least issues I'm running into.

The first I mentioned in the last comment about the drop zone dodging the cursor.

The other issue is me assuming the data model holds up without sticking it into one firm parent structure that feeds top down. I've run into a lot of "object could not be cloned" errors followed by dzToConfig.get(...) is undefined. Not sure the exact meaning of those but I'm sure the transition of DND 'items' from one data structure (cards) to (subtask) is messing with some of the library expectations depending on the timing or exacted place dropped.   

As a curiosity, I initiated two lists pointing at the same "cards" data. I'm seeing a lot of similar behaviour as I observed in my application. 

I'm going to move on to other priorities and touch on this again in the future. I think the answer to the original question is, yes it is possible to nest the same type DND zone. The functionality of anyone's given implementation may very. I noticed that increasing margins and dragging up increased the chance of being able to drop into the same type child zone. That said, I'll close this issue and maybe open up something more specific if need be about the practicality of this type of behaviour.
